src/imagery/i.eodag/i.eodag.py

```diff
@@ -530,7 +531,91 @@ def dates_to_iso_format():
     options["end"] = end_date
 
 
-def filter_result(search_result, geometry, **kwargs):
+def parse_query(query=None):
+    """Parse query string.
+
+    :param query: WKT String with the geometry to filter with respect to
+    :type query: str
+
+    :returns: A dictionary of queryables as keys, and a list of tuples,
+              each tuple consists of a queryable value and an operator,
+              as the dictionary values.
+              Dict["queryable", List(Tuple("queryable_value", "operator"))]
+    :rtype: Dict[str, List(Tuple(str, str))]
+
+    """
+    VALID_OPERATORS = ["eq", "ne", "ge", "gt", "le", "lt"]
+    DEFAULT_OPERATOR = "eq"
+    query_dicts = {}
+    if query is None:
+        return query_dicts
+    for parameter in map(str.strip, options["query"].split(",")):
+        if parameter == "":
+            continue
+        try:
+            key, values = map(str.strip, parameter.split("="))
+        except Exception as e:
+            gs.debug(e)
+            gs.fatal(_("Queryable <{}> could not be parsed".format(parameter)))
+        if key == "start":
+            try:
+                start_date = normalize_time(values)
+                query_dicts["start"] = [(start_date, DEFAULT_OPERATOR)]
+            except Exception as e:
+                gs.debug(e)
+                gs.fatal(
+                    _(
+                        "Queryable <{}> could not be parsed\nDate must be ISO formated".format(
+                            parameter
+                        )
+                    )
+                )
+            continue
+        if key == "end":
+            try:
+                end_date = normalize_time(values)
+                query_dicts["end"] = [(end_date, DEFAULT_OPERATOR)]
+            except Exception as e:
+                gs.debug(e)
+                gs.fatal(
+                    _(
+                        "Queryable <{}> could not be parsed\nDate must be ISO formated".format(
+                            parameter
+                        )
+                    )
+                )
+            continue
+        operator = None
+        values_operators = []
+        for value in map(str.strip, values.split("|")):
+            if value == "":
+                continue
+            if value.find("\\") != -1:
```

Comment:
The `\` is a bit awkward choice as it is used for masking and may make it more difficult to handle on command line. Another singlecharacter separator like `|`, or `#`, or `;` would be probably a bit more appropriate....

Reply:
I will change it to `;`.